Userland/Games/Chess/main.cpp

```diff
@@ -119,7 +119,9 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     TRY(game_menu->try_add_action(show_available_moves_action));
     TRY(game_menu->try_add_separator());
 
-    TRY(game_menu->try_add_action(GUI::CommonActions::make_quit_action([](auto&) {
+    TRY(game_menu->try_add_action(GUI::CommonActions::make_quit_action([&](auto&) {
+        // Ensure that a quit command is sent to the engine prior to application exit.
```

Comment:
Why do we need that?
I'm not even sure that it works when clicking on the top right close button

Reply:
I've removed this since it clearly doesn't fire reliably. Its not strictly necessary for our engine, as it will exit when it becomes disconnected from the GUI. We will need to implement a mechanism to ensure `quit` is called when we have 3rd party engine support,.

I'll fix it when it becomes a problem.

Reply:
> We will need to implement a mechanism to ensure quit is called when we have 3rd party engine support.

I'm really not sure about that, not exiting on broken pipe is really an engine issue IMO. And I'm by no mean an UCI expert, but at least `stockfish` seems to properly exit on EOF.